service/source.go

```diff
@@ -510,3 +462,59 @@ func (g *AutoParameterGenerator) enum(index []int, typ reflect.Type, f func(inde
 	}
 	return nil
 }
+
+// AutoParameterConfig contains configs of AutoParameter.
+type AutoParameterConfig map[AutoParameterConfigKey]string
+
+// AutoParameterConfigKey is the key of AutoParameterConfig.
+type AutoParameterConfigKey string
+
+const (
+	// AutoParameterConfigKeyDefaultValue is the key of default value.
+	AutoParameterConfigKeyDefaultValue AutoParameterConfigKey = "default"
+	// AutoParameterConfigKeyAnonymous is the prefix of anonymous configs.
+	AutoParameterConfigKeyAnonymous AutoParameterConfigKey = " anonymous"
+)
+
+// Get gets value of a config key.
+func (f AutoParameterConfig) Get(key AutoParameterConfigKey) string {
+	return f[key]
+}
+
+// Set sets value for a config key.
+func (f AutoParameterConfig) Set(key AutoParameterConfigKey, value string) {
+	f[key] = value
+}
+
+// ParseAutoParameterTag parse the tag of AutoParameter.
+func ParseAutoParameterTag(tag string) (source definition.Source, name string, apc AutoParameterConfig, err error) {
+	apc = make(AutoParameterConfig)
+	// This operation is problematic.
```

Comment:
this method hard-code quite a few things, which are actually user-facing (part of the framework API)?

Reply:
Yes. This method parses field tags from `Auto` parameter.